Testing powerups

FlXlT opened the pull request:
Tested the powerup package to a coverage of about 80%.

Added an abstract classe which is extended by the powerup pickups.
For the powerups I added an abstract PlayerElementTest class.

Issue: #267

FlXlT committed:
Added initial powerup testing class.

FlXlT committed:
Merge remote-tracking branch 'origin/release' into testing-powerups

FlXlT committed:
Added abstract PlayerElementTest

<testing

FlXlT committed:
Added test intitial test classes.

<testing

Commit:
Make fix

<organisation

FlXlT committed:
ChiliChickenTest updated to 70% coverage

<testing

FlXlT committed:
Finished the powerup pickup tests

Reached about 90% coverage

<testing

FlXlT committed:
Finshed testing ChiliChicken and TurtleTaco

The powerup package now has a coverage of 77%.
Also added JavaDoc.

<testing,documentation

FlXlT committed:
Merge remote-tracking branch 'origin/release' into testing-powerups

NielsWarnars commented:
Please run Maven site and fix the CheckStyle/PMD/FindBugs issues. Till those are fixed no merge will occur.
<review

NielsWarnars commented on src/test/java/nl/tudelft/scrumbledore/level/powerup/TurtleTacoPickUpTest.java:
```diff
@@ -0,0 +1,80 @@
+import nl.tudelft.scrumbledore.sprite.Sprite;
+
+/**
+ * Tests the TurtleTacoPickUp class which makes use of the LevelElementTest class.
```
I don't think it makes use of the LevelElementTest class.
<review

FlXlT committed:
Added missing JavaDoc.

Commit:
Fix CheckStyle and PMD suggestions

Fixes code quality suggestions neglected by the original author.
<organisation

Commit:
Merge branch 'release' into testing-powerups

Commit:
Suppress too many methods warning

Suppress too many methods warning in the PlayerActionsLevelModifier test suite.
<organisation
